File: game/bin/tools/process_defs/resources/scripts/GenerateJavaScript/templates/DefsDigest.cpp
#include "DefsDigest.hpp"

#include "connection/entity_def_constants.hpp"

namespace DefsDigest
{

BW::EntityDefConstants g_entityDefConstants(
		BW::MD5::Digest( "{{ constants.digest }}" ),
		{{ constants.maxExposedClientMethodCount }}, // numClientMethods
		{{ constants.maxExposedBaseMethodCount }}, // numBaseMethods
		{{ constants.maxExposedCellMethodCount }} ); // numCellMethods

const BW::EntityDefConstants & constants()
{
	return g_entityDefConstants;
}

} // namespace DefsDigest


// DefsDigest.cpp

File: game/bin/tools/process_defs/resources/scripts/GenerateCSharp/templates/DefsDigest.cs
using System;
using System.Collections.Generic;
using System.Text;

{% if generatedNamespace != "BW" -%}
using BW;{% endif %}

namespace {{ generatedNamespace }}
{
public static class DefsDigest
{
	static public EntityDefConstants get()
	{
		return new EntityDefConstants( "{{ digest }}",
			{{ constants.maxExposedClientMethodCount }}, // numClientMethods
			{{ constants.maxExposedBaseMethodCount }}, // numBaseMethods
			{{ constants.maxExposedCellMethodCount }} ); // numCellMethods
	}
}
} // namespace {{ generatedNamespace }}


